atlite/datasets/cmip.py

```diff
@@ -0,0 +1,259 @@
+    ds = retrieve_data(
+        esgf_params,
+        coords,
+        variables=["mrro"],
+        **retrieval_params,
+    )
+    ds = _rename_and_fix_coords(ds, cutout.dt)
+    ds = ds.rename({"mrro": "runoff"})
+    return ds
+
+
+def sanitize_runoff(ds):
+    """Sanitize retrieved runoff data."""
+    ds["runoff"] = ds["runoff"].clip(min=0.0)
+    return ds
+
+
+def get_data_influx(esgf_params, cutout, **retrieval_params):
+    """Get influx data for given retrieval parameters."""
+    coords = cutout.coords
+    ds = retrieve_data(
+        esgf_params,
+        coords,
+        variables=["rsds", "rsus"],
+        **retrieval_params,
+    )
+
+    ds = _rename_and_fix_coords(ds, cutout.dt)
+
+    ds = ds.rename({"rsds": "influx", "rsus": "outflux"})
+
+    return ds
+
+
+def sanitize_inflow(ds):
+    """Sanitize retrieved inflow data."""
+    ds["influx"] = ds["influx"].clip(min=0.0)
+    return ds
+
+
+def get_data_temperature(esgf_params, cutout, **retrieval_params):
+    """Get temperature for given retrieval parameters."""
+    coords = cutout.coords
+    ds = retrieve_data(esgf_params, coords, variables=["tas"], **retrieval_params)
+
+    ds = _rename_and_fix_coords(ds, cutout.dt)
+    ds = ds.rename({"tas": "temperature"})
+    ds = ds.drop_vars("height")
+
+    return ds
+
+
+def get_data_wind(esgf_params, cutout, **retrieval_params):
+    """Get wind for given retrieval parameters"""
+
+    coords = cutout.coords
+    ds = retrieve_data(esgf_params, coords, ["sfcWind"], **retrieval_params)
+    ds = _rename_and_fix_coords(ds, cutout.dt)
+    ds = ds.rename({"sfcWind": "wnd{:0d}m".format(int(ds.sfcWind.height.values))})
+    ds = ds.drop_vars("height")
+    return ds
+
+
+def retrieve_data(esgf_params, coords, variables, chunks=None, tmpdir=None, lock=None):
+    """
+    Download data from egsf database
+
+    """
+    time = coords["time"].to_index()
+    years = time.year.unique()
+    dsets = []
+    if lock is None:
+        lock = nullcontext()
+    with lock:
+        for variable in variables:
+            esgf_params["variable"] = variable
+            search_results = search_ESGF(esgf_params)
+            files = [
+                f.opendap_url
+                for f in search_results
+                if int(f.opendap_url.split("_")[-1][:4]) in years
+            ]
+
+            dsets.append(xr.open_mfdataset(files, chunks=chunks, concat_dim=["time"]))
+    ds = xr.merge(dsets)
+
+    ds.attrs = {**ds.attrs, **esgf_params}
+
+    return ds
+
+
+def _rename_and_fix_coords(ds, dt, add_lon_lat=True, add_ctime=False):
+    """Rename 'longitude' and 'latitude' columns to 'x' and 'y' and fix roundings.
+
+    Optionally (add_lon_lat, default:True) preserves latitude and longitude
+    columns as 'lat' and 'lon'.
+
+    CMIP specifics; shif the longitude from 0..360 to -180..180. In addition
+    CMIP sometimes specify the time in the center of the output intervall this shifted
+    to the beginning.
```

Comment:
I would rather shift it to the end of the period, see https://atlite.readthedocs.io/en/master/conventions.html#time-points